### ISSUE TITLE
Notifications has stopped showing up (Manjaro 5.12.1-2 KDE Plasma 5.21.4)

### ISSUE THREAD

**Issue by tnilzon** [hluk/CopyQ#1664]
**Describe the bug**
Notifications has stopped showing up.

**To Reproduce**
1. Go to ''File/Preferences/Notifications"
2. Activate Notifications with the desired options.

**Expected behavior**
A desktop notification shall show up on the desktop displaying the clipboard content when it is updated.

**Version, OS and Environment**
CopyQ version 4.7.1
Operating System: Manjaro Linux
KDE Plasma Version: 5.21.4
KDE Frameworks Version: 5.81.0
Qt Version: 5.15.2
Kernel Version: 5.12.1-2-MANJARO
OS Type: 64-bit
Graphics Platform: X11
Processors: 8 × Intel® Core™ i7-9700K CPU @ 3.60GHz
Memory: 31,2 GiB of RAM
Graphics Processor: Mesa Intel® UHD Graphics 630

**Additional context**
This error occurred after the latest Manjaro (and KDE) update to versions as listed above. Before the update everything was OK.

**Comment by hluk** [comment]
> CopyQ version 4.7.1

There is no such CopyQ version yet. :)

CopyQ 4.0.0 uses native notifications (whatever the desktop environment provides).

4.1.0 added an option to use the old builtin notification system instead (disable "Use native notifications" option in Notification tab).

**Comment by tnilzon** [comment]
Oh, sorry about CopyQ version. That was a miss type in my post. It should be v4.1.0
I have tried the settings both with and without "use native notifications".
However, the result is still the same, the notifications do not work for me.

**Comment by hluk** [comment]
Can you test it with the following command?

```
copyq 'notification(".title", "TEST", ".message", "test", ".time", 5000)'
```

**Comment by tnilzon** [comment]
copyq 'notification(".title", "TEST", ".message", "test", ".time", 5000)'

Tried that, nothing happened. Nothing showed up.

**Comment by hluk** [comment]
Can you try launching copyq again with verbose/debug logs?

```bash
copyq exit

# Show logs from notification system (less verbose)...
QT_LOGGING_RULES='kf.notifications.debug=true' copyq

# ...or all logs from Qt and the app (very verbose)
QT_LOGGING_RULES='*.debug=true' COPYQ_LOG_LEVEL=DEBUG copyq
```

**Comment by tnilzon** [comment]
Yes, I have now done that. Are there any log-file stored somewhere (I can't find any)?

**Comment by hluk** [comment]
You can list logs with:

```
copyq logs
```

Or print log file (there are also older log files in the same directory `copyq.log.1`, `copyq.log.2` etc):

```
copyq info log
```

**Comment by tnilzon** [comment]
Here are two log files:
http://u.pc.cd/Kf8otalK
http://u.pc.cd/YT6ctalK

**Comment by hluk** [comment]
I cannot find any log lines related to notification. Looks like the notification system is not invoked at all.

Can you share the settings in Notification tab?

**Comment by tnilzon** [comment]
![image](https://user-images.githubusercontent.com/6920015/119222434-a4868d00-baf4-11eb-9baf-3a2797f3139b.png)

**Comment by hluk** [comment]
Do any notifications from other programs show up? E.g. `notify-send title message`.

Can you check what is printed with the following commands?

```bash
copyq exit
QT_LOGGING_RULES='kf.notifications.*.debug=true' COPYQ_LOG_LEVEL=DEBUG copyq --start-server
copyq config native_notifications true
copyq 'serverLog("BEGIN_TEST"); notification(".title", "TEST", ".message", "test", ".time", 5000); serverLog("END_TEST")'
copyq logs | sed -n '/BEGIN_TEST/,/END_TEST/{p}'
```

There should be something like `[kf.notifications] QtDebug: Calling notify on "Popup"`.

**Comment by tnilzon** [comment]
notify-send works as normally.

Your command in terminal give me this:

[tomas@tomas-pc ~]$ copyq exit
QT_LOGGING_RULES='kf.notifications.*.debug=true' COPYQ_LOG_LEVEL=DEBUG copyq --start-server
copyq config native_notifications true
copyq 'serverLog("BEGIN_TEST"); notification(".title", "TEST", ".message", "test", ".time", 5000); serverLog("END_TEST")'
copyq logs | sed -n '/BEGIN_TEST/,/END_TEST/{p}'
Terminating server.
true
CopyQ Note [2021-05-23 08:37:20.340] <Server-180611>: BEGIN_TEST
CopyQ Note [2021-05-23 08:37:20.362] <Server-180611>: END_TEST

**Comment by hluk** [comment]
It looks like native (and possibly builtin) notifications are not invoked at all from CopyQ. I'm running out of ideas. 🤔

Can you share output of following commands?

```bash
copyq version
copyq info
```

**Comment by tnilzon** [comment]
[tomas@tomas-pc ~]$ copyq version
CopyQ Clipboard Manager v4.1.0
Qt: 5.15.2
KNotifications: 5.81.0
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: Manjaro Linux

[tomas@tomas-pc ~]$ copyq info
config: /home/tomas/.config/copyq/copyq.conf
desktop: /usr/share/applications/com.github.hluk.copyq.desktop
exe: /usr/bin/copyq
has-global-shortcuts: 1
has-mouse-selection: 1
icons: /usr/share/icons/hicolor/scalable/apps/copyq
log: /home/tomas/.local/share/copyq/copyq/copyq.log
platform: Linux/X11
plugins: /usr/lib/copyq/plugins
themes: /usr/share/copyq/themes
themes(custom): 
translations: /usr/share/copyq/translations
translations(custom):

**Comment by tnilzon** [comment]
I have now also tried CopyQ v4.1.0 on my Windows 10 PC. The same issue appears also there, no notifications appears.

**Comment by hluk** [comment]
> I have now also tried CopyQ v4.1.0 on my Windows 10 PC. The same issue exists also there, no notifications appears.

Hmm, this must be some configuration in CopyQ.

Can you try setting some positive value for "Interval in seconds to display notifications" option?

**Comment by tnilzon** [comment]
Can you try setting some positive value for "Interval in seconds to display notifications" option?

No, that makes no difference.

**Comment by Chealer** [comment]
@tnilzon: 5.12.1 is your version of Linux, there is no such version of Manjaro.

Does this persist on Manjaro 26 or above? And if so, how did you install CopyQ?
